### DISCUSSION TITLE
Add private packages support for maven-wrapper updates

### DISCUSSION THREAD

**Discussion by Churro** [renovatebot/renovate#25791] [Suggest an Idea]
### Tell us more.

renovate currently lacks support for passing package manger credentials to `mvn` for artifact updates. This causes maven-wrapper updates to fail, as the call to `./mvnw wrapper:wrapper` isn't provided the credentials it needs to resolve private dependencies.

Related discussion: https://github.com/renovatebot/renovate/discussions/25066

**Proposed Solution**

Extend https://docs.renovatebot.com/getting-started/private-packages/#maven to support for username / password and forward it to `mvn`. A concrete way could be to realize an analogous solution to `nuget` [here](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/nuget/config-formatter.ts):

- Write matching host rules into a temporary `settings.xml` that will be then picked up by `mvn` automatically. 
- Alternatively it may be possible to simply map the username to the environment variable `MVNW_USERNAME` and password to `MVNW_PASSWORD` and ensure they are forwarded when invoking `./mvnw wrapper:wrapper`.

**Comment by rarkins** [comment]
Converted to issue #25805
